clippy_lints/src/lib.rs

```diff
@@ -911,6 +912,7 @@ pub fn register_plugins(store: &mut rustc_lint::LintStore, sess: &Session, conf:
     store.register_late_pass(|_| Box::new(size_of_ref::SizeOfRef));
     store.register_late_pass(|_| Box::new(multiple_unsafe_ops_per_block::MultipleUnsafeOpsPerBlock));
     store.register_late_pass(|_| Box::new(extra_unused_type_parameters::ExtraUnusedTypeParameters));
+    store.register_pre_expansion_pass(|| Box::<missing_assert_message::MissingAssertMessage>::default());
```

Comment:
Unfortunately we don't want to be adding more pre expansion passes because it comes with a good number of bugs - #6610, it would want to be a late pass

We do have some utilities that would help out there: [`find_assert_args`](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/macros/fn.find_assert_args.html) and [`find_assert_eq_args`](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/macros/fn.find_assert_eq_args.html). Usage would be similar to

https://github.com/rust-lang/rust-clippy/blob/6444621c101151b49a5e7c9854d602483000e337/clippy_lints/src/assertions_on_constants.rs#L34-L40

Reply:
We started with a late pass and `find_assert_args` functions, but I think there was a recent change in the compiler and assert macros expand to something else than `find_assert_args` expects. So those functions don't return the correct `PanicExpn`, we tried to fix it but couldn't do it :) I'll give it another go!

Reply:
So it seems like starting from Rust 1.52 `assert_{eq,ne}` macros use `panicking::assert_failed` instead of `panic!` (See `assert_eq` in [1.52](https://github.com/rust-lang/rust/blob/1.52.0/library/core/src/macros/mod.rs#L35-L62) and in [1.51](https://github.com/rust-lang/rust/blob/1.51.0/library/core/src/macros/mod.rs#L56-L86)).

https://github.com/rust-lang/rust-clippy/pull/10362/commits/a2502da1c4f6dee0f0444072385588827209671c addresses Rust 1.52 changes
https://github.com/rust-lang/rust-clippy/pull/10362/commits/569fa96915bfddf8724b5141382b3ca91393a858 converts this lint to a late lint pass